libarchive/archive_read_support_filter_gzip.c

```diff
@@ -214,11 +226,31 @@ gzip_bidder_bid(struct archive_read_filter_bidder *self,
 
 	(void)self; /* UNUSED */
 
-	if (peek_at_header(filter, &bits_checked))
+	if (peek_at_header(filter, &bits_checked, NULL))
 		return (bits_checked);
 	return (0);
 }
 
+static int
+gzip_read_header(struct archive_read_filter *self, struct archive_entry *entry)
+{
+	struct private_data *state;
+
+	state = (struct private_data *)self->data;
+
+	/* A mtime of 0 is considered invalid/missing. */
+	if (state->mtime != 0)
+		archive_entry_set_mtime(entry, state->mtime, 0);
+
+	/* If the name is available, extract it. */
+	if (state->name)
+		archive_entry_set_pathname(entry, state->name);
+
+	if (state->file_size != 0)
+		archive_entry_set_size(entry, state->file_size);
```

Comment:
Can this ever happen?  The `file_size` can only be read from the trailer, so I think it will never actually be set here?

Reply:
`file_size` is coming from the gzip metadata rather than the decompressed length, so it seems like in theory this could be anything.  we're relying on the compressor correctly filling this out.

there's also a comment in this file about "Verify the length" below ...

Reply:
(meant to do a bit more research before posting, but hit the wrong button)

i played a bit more and it looks like it's working by happy accident.  when the archive is small enough, `gzip_filter_read` hits `Z_STREAM_END` right away which means `file_size` is set before this callback is processed.  obviously that doesn't fly for larger archives.

i'll pull out `file_size` processing for now from this commit until i can find a better way of accessing the data.  if the stream is seekable, i should be able to pull it out ...